addons/base_import/models/base_import.py

```diff
@@ -476,107 +554,208 @@ def _try_match_date_time(self, preview_values, options):
+        type_fields = {}
         if preview:
-            for column in range(0, len(preview[0])):
-                preview_values = [value[column].strip() for value in preview]
-                type_field = self._try_match_column(preview_values, options)
-                type_fields.append(type_field)
+            for column_index in range(0, len(headers)):
+                preview_values = [value[column_index].strip() for value in preview if len(value) > column_index]
+                type_field = self._get_most_likely_header_types(preview_values, options)
+                type_fields[(column_index, headers[column_index])] = type_field
         return type_fields
 
-    def _match_header(self, header, fields, options):
-        """ Attempts to match a given header to a field of the
-            imported model.
-
-            :param str header: header name from the CSV file
-            :param fields:
-            :param dict options:
-            :returns: an empty list if the header couldn't be matched, or
-                      all the fields to traverse
-            :rtype: list(Field)
+    def _get_mapping_suggestion(self, header, fields_tree, header_types):
+        """ Attempts to match a given header to a field of the imported model.
+
+            We can distinguish 2 types of header format:
+                - simple header string that aim to directly match a field of the target model
+                    E.g.: "lead_id" or "Opportunities" or "description"
+                - composed '/' joined header string that aim to match a field of a relation field of the target model (= subfield)
+                    E.g.: 'lead_id/description' aim to match the field 'description' of the field lead_id.
+            When returning result, the header is transformed into a field path to ease further treatments.
+                - ["lead_id"] for simple case = simple matching
+                - ["lead_id", "description"] for composed case = hierarchy matching
+
+            First try to make an exact match with all the field of the model.
+            If no exact match can be found, try the "fuzzy match" (using word distance - Levenshtein distance)
+            on the most likely field types. For subfields, use the same logic.
+
+            Levenshtein distance is a score between 0 and 1 to express the distance between two char strings where:
+                0 = exact match
+                1 = completely different strings
+
+            In order to keep only one column matched per field, we return the distance. That distance will be used
+            during the deduplicate process and only the mapping with the smallest distance will be kept in case of
+            multiple mapping on the same field.
+            Note that we don't need to return the distance in case of hierachy mapping as we consider that
+            as advanced behaviour. The deduplicate process will ignore hierarchy mapping.
```

Comment:
"The user will have to manually select on which field he wants to map what in case of mapping duplicates for sub-fields".
(Without that it feels like it "will not work").